**ECTA Recruiting Data Project**

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as mpl
pd.set_option('display.max_rows', None)

In [3]:
df1 = pd.read_csv("ApplicantData1.csv")

In [4]:
df2 = pd.read_csv("ApplicantData2.csv")

In [5]:
df1.columns

Index(['Application ID', 'Candidate ID', 'Application Date',
       'Recruiter User Sys ID', 'Recruiter First Name', 'Recruiter Last Name',
       'Level / Grade', 'Is Internal', 'Gender (TA maintained)', 'Job Req ID',
       'Application Country', 'Job Title', 'Application Status',
       'Position Number', 'Date Created', 'Template ID', 'Source Details',
       'Source', 'Hiring Manager User Sys ID', 'Candidate Country/Region',
       'Approved Date', 'Hired On', 'Current / Latest Level of Education',
       'Current / Latest Major', 'Current / Latest University or School',
       'If Other Major', 'If Other University or School', 'Contract type'],
      dtype='object')

In [6]:
df2.columns

Index(['Job Req ID', 'Application ID', 'Application Status', 'Created Date',
       'Candidate Country/Region', 'Application Country',
       'Current / Latest Level of Education', 'Current / Latest Major',
       'Current / Latest University or School', 'If Other Major',
       'If Other University or School', 'Source', 'Source Details'],
      dtype='object')

In [7]:
combined = pd.merge(df1, df2, how='left', on=['Application ID', 'Current / Latest University or School', 'Application Country'
                                             , 'If Other Major', 'If Other University or School', 'Current / Latest Level of Education'
                                             , 'Current / Latest Major','Candidate Country/Region', 'Current / Latest Major',
                                             'Application Status', 'Source', 'Source Details', 'Job Req ID'])

In [8]:
combined.shape

(25992, 29)

In [9]:
combined.columns

Index(['Application ID', 'Candidate ID', 'Application Date',
       'Recruiter User Sys ID', 'Recruiter First Name', 'Recruiter Last Name',
       'Level / Grade', 'Is Internal', 'Gender (TA maintained)', 'Job Req ID',
       'Application Country', 'Job Title', 'Application Status',
       'Position Number', 'Date Created', 'Template ID', 'Source Details',
       'Source', 'Hiring Manager User Sys ID', 'Candidate Country/Region',
       'Approved Date', 'Hired On', 'Current / Latest Level of Education',
       'Current / Latest Major', 'Current / Latest University or School',
       'If Other Major', 'If Other University or School', 'Contract type',
       'Created Date'],
      dtype='object')

In [10]:
combined.head(3)

Application ID  Candidate ID Application Date  Recruiter User Sys ID  \
0         6504753       4451438        9/21/2021                 545796   
1         7126456       4475029         4/6/2022                 545796   
2         7213676       3831108         5/5/2022                 545796   

  Recruiter First Name Recruiter Last Name          Level / Grade Is Internal  \
0                Nelly          Hufschmitt  Not a graded position          No   
1                Nelly          Hufschmitt  Not a graded position          No   
2                Nelly          Hufschmitt  Not a graded position          No   

  Gender (TA maintained)  Job Req ID  ... Candidate Country/Region  \
0                    NaN      355808  ...                       FR   
1                    NaN      355769  ...                       FR   
2                    NaN      355650  ...                       FR   

    Approved Date Hired On  Current / Latest Level of Education  \
0  8/3/2021 14:33      NaN    High School Diploma or Equivalent   
1  8/3/2021 14:29      NaN                            Bachelors   
2  8/3/2021 14:12      NaN                            Bachelors   

  Current / Latest Major  Current / Latest University or School  \
0            Other Major             Other University or School   
1             Accounting             Other University or School   
2              Languages             Other University or School   

                                      If Other Major  \
0  commercialisation des produits et services spo...   
1                                                NaN   
2                                                NaN   

  If Other University or School  Contract type Created Date  
0                   ILEPS Cergy         Intern   11/24/2021  
1                           NaN         Intern     4/6/2022  
2                           NaN         Intern     5/5/2022  

[3 rows x 29 columns]

In [11]:
conditions = ['US', 'CN'] #conditionals for canada

usinterns = combined[combined['Candidate Country/Region'] == "US"]

# usmasters = usinterns[usinterns['Current / Latest Level of Education'].isin(options)]  **tweak to get canada in too if we want :D

In [12]:
usinterns.shape

(9244, 29)

In [13]:
usinterns['Current / Latest University or School'].value_counts()
#Remove .head(3) to see all

Other University or School                                                             6256
University of Oregon                                                                    369
Arizona State University (ASU)                                                          307
University of California                                                                277
University of Texas                                                                     238
University of Washington                                                                216
Portland State University                                                               189
University of Southern California                                                       174
Rutgers University                                                                      135
Northwestern University                                                                 116
North Carolina State University (NC State University)                           

In [14]:
usinterns['If Other University or School'].value_counts().head(3)
#Remove .head(3) to see all

Brigham Young University     103
University of Portland        93
Michigan State University     91
Name: If Other University or School, dtype: int64

In [15]:
usinterns.head(3)

Application ID  Candidate ID Application Date  Recruiter User Sys ID  \
4         6971340       3136252        2/28/2022                 751078   
5         6994274       1511962        2/26/2022                 732021   
6         6999629       3832077         3/1/2022                 732021   

  Recruiter First Name Recruiter Last Name          Level / Grade Is Internal  \
4                 Eric              Bailey  Not a graded position          No   
5             Angelica           Gutierrez  Not a graded position          No   
6             Angelica           Gutierrez  Not a graded position          No   

  Gender (TA maintained)  Job Req ID  ... Candidate Country/Region  \
4                    NaN      406591  ...                       US   
5                    NaN      411588  ...                       US   
6                    NaN      411588  ...                       US   

      Approved Date Hired On  Current / Latest Level of Education  \
4  12/16/2021 19:36      NaN    High School Diploma or Equivalent   
5  12/29/2021 23:51      NaN                              Masters   
6  12/29/2021 23:51      NaN                            Bachelors   

  Current / Latest Major  Current / Latest University or School  \
4             Art/Design             Other University or School   
5              Education             Other University or School   
6             Psychology             Other University or School   

  If Other Major                If Other University or School  Contract type  \
4            NaN                         Stockton Univerisity         Intern   
5            NaN                     College of Staten Island         Intern   
6            NaN  California State University Dominguez Hills         Intern   

  Created Date  
4     4/2/2022  
5     4/8/2022  
6     4/9/2022  

[3 rows x 29 columns]

In [16]:
usinterns['Application Status'].value_counts()

#Total US Intern App Results

Rejected - automatic email notification enabled    8033
Rejected                                            776
Requisition Closed                                  232
Hire                                                 67
Withdrawn By Candidate                               47
Auto Disqualified                                    47
Hired On Other Requisition                           33
Recruiter Interview                                   5
Default                                               2
Deleted On Demand By Candidate                        1
Offer                                                 1
Name: Application Status, dtype: int64

In [17]:
options = ['Masters', 'MBA'] 
    
# selecting rows based on condition 
usmasters = usinterns[usinterns['Current / Latest Level of Education'].isin(options)] 


In [18]:
usmasters.head(5)
usmasters['Application Status'].value_counts()

Rejected - automatic email notification enabled    2705
Rejected                                            424
Requisition Closed                                   66
Auto Disqualified                                    18
Hire                                                 17
Withdrawn By Candidate                               14
Hired On Other Requisition                           13
Default                                               1
Recruiter Interview                                   1
Name: Application Status, dtype: int64

In [19]:
usmasters.columns

Index(['Application ID', 'Candidate ID', 'Application Date',
       'Recruiter User Sys ID', 'Recruiter First Name', 'Recruiter Last Name',
       'Level / Grade', 'Is Internal', 'Gender (TA maintained)', 'Job Req ID',
       'Application Country', 'Job Title', 'Application Status',
       'Position Number', 'Date Created', 'Template ID', 'Source Details',
       'Source', 'Hiring Manager User Sys ID', 'Candidate Country/Region',
       'Approved Date', 'Hired On', 'Current / Latest Level of Education',
       'Current / Latest Major', 'Current / Latest University or School',
       'If Other Major', 'If Other University or School', 'Contract type',
       'Created Date'],
      dtype='object')

In [20]:
usmasters_simple = usmasters[['Recruiter First Name', 'Recruiter Last Name', 'Application Country', 
                              'Job Title','Application Status','Candidate Country/Region', 
                              'Current / Latest Level of Education','Current / Latest Major',
                              'Current / Latest University or School','If Other Major',
                              'If Other University or School']].copy()
                        
    

In [21]:
usmasters_simple.shape

(3259, 11)

In [22]:
usmasters_simple['Application Status'].value_counts()

Rejected - automatic email notification enabled    2705
Rejected                                            424
Requisition Closed                                   66
Auto Disqualified                                    18
Hire                                                 17
Withdrawn By Candidate                               14
Hired On Other Requisition                           13
Default                                               1
Recruiter Interview                                   1
Name: Application Status, dtype: int64

3259 Total Masters/MBA Applications

Only 14 withdrawn by candidate.

**Overall ratio = 14/3259 = 0.00429579626**

By School is below:

In [23]:
renege_grads = usmasters_simple.loc[usmasters_simple['Application Status'] == 'Withdrawn By Candidate']

renege_grads

Recruiter First Name Recruiter Last Name Application Country  \
5374              Angelica           Gutierrez                  US   
7332                Rafael   de Castro Camargo                  DE   
13777               Rafael   de Castro Camargo                  DE   
14687               Rafael   de Castro Camargo                  US   
16958                Vicki                  Ng                  US   
17389               Rafael   de Castro Camargo                  DE   
17448             Angelica           Gutierrez                  US   
17460             Angelica           Gutierrez                  US   
17461             Angelica           Gutierrez                  US   
19369                  Meg            Anderson                  US   
23144              America               Perez                  CL   
23284               Rafael   de Castro Camargo                  DE   
23328             Angelica           Gutierrez                  US   
23917               Rafael   de Castro Camargo                  CH   

                                               Job Title  \
5374    2022 US Internship - Supply Chain Product Supply   
7332   MBA Summer Internship - Business Development &...   
13777  MBA Summer Internship - Human Resources Manage...   
14687  2022 US Internship MBA - eCommerce Consumer Ex...   
16958  2022 US Internship MBA - North America Supply ...   
17389  MBA Summer Internship - Digital Partner Commer...   
17448   2022 US Internship - Supply Chain Product Supply   
17460  2022 US Internship - Originals Product Marketi...   
17461  2022 US Internship - Originals Product Marketi...   
19369   2022 US Internship - Global Digital Partnerships   
23144                                      Intern Reebok   
23284  MBA Summer Internship - Digital Strategy HZO (...   
23328  2022 US Internship -  Originals Culture Collab...   
23917  MBA Summer Internship - Global Operations Stra...   

           Application Status Candidate Country/Region  \
5374   Withdrawn By Candidate                       US   
7332   Withdrawn By Candidate                       US   
13777  Withdrawn By Candidate                       US   
14687  Withdrawn By Candidate                       US   
16958  Withdrawn By Candidate                       US   
17389  Withdrawn By Candidate                       US   
17448  Withdrawn By Candidate                       US   
17460  Withdrawn By Candidate                       US   
17461  Withdrawn By Candidate                       US   
19369  Withdrawn By Candidate                       US   
23144  Withdrawn By Candidate                       US   
23284  Withdrawn By Candidate                       US   
23328  Withdrawn By Candidate                       US   
23917  Withdrawn By Candidate                       US   

      Current / Latest Level of Education   Current / Latest Major  \
5374                                  MBA  Business administration   
7332                                  MBA              Other Major   
13777                             Masters          Human Resources   
14687                                 MBA  Business administration   
16958                                 MBA  Business administration   
17389                                 MBA              Other Major   
17448                             Masters              Other Major   
17460                             Masters    Sports/Sports science   
17461                             Masters    Sports/Sports science   
19369                             Masters              Other Major   
23144                                 MBA    Sports/Sports science   
23284                                 MBA              Other Major   
23328                             Masters              Other Major   
23917                                 MBA  Business administration   

       Current / Latest University or School  \
5374                    University of Oregon   
7332              Other University o

**Of the 14, 4 were University of Oregon, 3 were Cornell, 2 were UT. All others were single isolated cases.**